src/librustc/query/mod.rs

```diff
@@ -421,9 +421,16 @@ rustc_queries! {
     BorrowChecking {
         /// Borrow-checks the function body. If this is a closure, returns
         /// additional requirements that the closure's creator must verify.
-        query mir_borrowck(key: DefId) -> mir::BorrowCheckResult<'tcx> {
+        query mir_borrowck(key: DefId) -> &'tcx mir::BorrowCheckResult<'tcx> {
             desc { |tcx| "borrow-checking `{}`", tcx.def_path_str(key) }
-            cache_on_disk_if(tcx, _) { key.is_local() && tcx.is_closure(key) }
+            cache_on_disk_if { key.is_local() }
```

Comment:
Why are you caching non-closure keys?

Reply:
Because it's no longer just closures that can have interesting borrow checking results. I didn't realize that this could depend on the query's result though, I can change it to that if that wouldn't cause problems.

Reply:
I think you can look at the values, yes. What's looking at non-closure borrow checking results now?

Reply:
`type_of` for opaque types is looking at it.